org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/Point.java

```diff
@@ -133,4 +135,21 @@ public int getIndex(String name) {
 		}
 		return indices;
 	}
+
+	private static final String VERTEX = "([a-zA-Z0-9_])+\\((\\d+)\\)=([-+]?[0-9]*\\.?[0-9]+)";
+	private static final Pattern POSITION = Pattern.compile(VERTEX + ", " + VERTEX);
+
+	/**
+	 * Parse a point from the toString() method into an instance of Point
+	 * @param asString
+	 * @return
+	 */
+	public static Point parse(String asString) {
+		Matcher m = POSITION.matcher(asString);
+		if (m.matches()) {
+			return new Point(m.group(4), Integer.parseInt(m.group(5)), Double.parseDouble(m.group(6)),
+					m.group(1), Integer.parseInt(m.group(2)), Double.parseDouble(m.group(3)));
+		}
+		throw new RuntimeException("Unparsable string" + asString);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)